corehq/apps/app_manager/suite_xml/xml_models.py

```diff
@@ -576,9 +609,12 @@ class Entry(OrderedXmlObject, XmlObject):
 
     form = StringField('form')
     post = NodeField('post', RemoteRequestPost)
+
+    # command and localized_command are mutually exclusive based on the app version
     command = NodeField('command', Command)
-    instances = NodeListField('instance', Instance)
+    localized_command = NodeField('command', LocalizedCommand)
```

Comment:
can you explain how this works?  I see that the suite generation code will set `command` to an instance of either `Command` or `LocalizedCommand` depending on the version, but how does having multiple attributes with the same name work here?  Do they both point to the same thing (`'command'`)?  Does that mean that `localized_command` will be populated even on older versions where the populated value is a `Command` instance?

Reply:
The Python objects are mostly designed to simplify accessing XML data from python. The docs say:

> When code accesses these fields on the object, the code evaluates the associated XPath expression and converts the data to a Python value. https://eulxml.readthedocs.io/en/latest/xmlmap.html

So when you access an attribute it will populate the python objects appropriately but it won't create XML that's not there unless you assign it a value. Here's an example:

```python
>>> Entry().serialize()
b'<entry/>'

>>> e = Entry()
>>> e.localized_command is None
True
>>> e.serialize()
b'<entry/>'
>>> e.command = Command()

# command and localized_command are different objects but under refer to the same XML node:
>>> e.command
<corehq.apps.app_manager.suite_xml.xml_models.Command at 0x7f7c68bdefa0>
>>> e.localized_command
<corehq.apps.app_manager.suite_xml.xml_models.LocalizedCommand at 0x7f7c6923abb0>

>>> e.command.node
<Element command at 0x7f7c5bbc3fc0>
>>> e.localized_command.node
<Element command at 0x7f7c5bbc3fc0>

```

Reply:
nice, thanks for laying that out, that's very helpful.  That's vaguely what I was imagining, but hadn't explored it much.